File: src/main/java/ar/edu/unq/spring/configuration/AppConfiguration.java
package ar.edu.unq.spring.configuration;

import ar.edu.unq.spring.service.impl.FakeRandomizer;
import ar.edu.unq.spring.service.impl.RealRandomizer;
import ar.edu.unq.spring.service.interfaces.Randomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;


@Configuration
public class AppConfiguration {

    @Primary
    @Bean(name = "RealRandomizer")
    public Randomizer Randomizer() {
        return new RealRandomizer();
    }

    @Bean(name = "FakeRandomizer")
    public Randomizer FakeRandomizer() {
        return new FakeRandomizer();
    }

    @Bean(name = "RandomizerMagico")
    public Randomizer AnotherFakeRandomizer() {
        return () -> Math.random() * 100 > 50 ? 100 : 0;
    }
}

File: src/main/kotlin/ar/edu/unq/spring/configuration/AppConfiguration.kt
package ar.edu.unq.spring.configuration

import ar.edu.unq.spring.service.transaction.SpringRelationalTransaction
import ar.edu.unq.spring.service.transaction.TransactionRunner
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

// los Beans se crean aca. Como Para este proyecto chico, solo tenemos componentes de Spring
// Ya spring se encarga de hacer el wiring, sin necesidad que declaremos los objetos aca
@Configuration
class AppConfiguration @Autowired constructor(
    private val springRelationalTransaction: SpringRelationalTransaction
) {
    @Bean
    fun configureTransactionRunner() {
        TransactionRunner.addTransaction(springRelationalTransaction)
    }

}